sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -338,8 +339,11 @@ class JacksonParser(
               UTF8String.fromBytes(writer.toByteArray)
           }
         }
-      // Reset back to the original configuration:
-      parser.setFeatureMask(oldFeature)
+      // Reset back to the original configuration using `~0` as the mask,
+      // which is a bitmask with all bits set, effectively allowing all features
+      // to be reset. This ensures that every feature is restored to its previous
+      // state as defined by `oldFeature`.
+      parser.overrideStdFeatures(oldFeature, ~0)
```

Comment:
can also use a seat mask of `-1` to restore the state, as long as all the bits are set to 1.